examples/custom_route_rules/terraform.tfvars.example

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) 2019, 2020 Oracle Corporation and/or affiliates.  All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl
+
+# provider identity parameters
+
+api_fingerprint = ""
+
+api_private_key_path = ""
+
+region = "us-phoenix-1"
+
+tenancy_id = ""
+
+user_id = ""
+
+# general oci parameters
+
+compartment_id = ""
+
+label_prefix = "dev"
+
+# vcn parameters
+
+create_drg = false
+
+internet_gateway_enabled = false
+
+nat_gateway_enabled = false
+
+service_gateway_enabled = false
+
+vcn_cidr = "10.0.0.0/16"
+
+vcn_dns_label = "vcn"
+
+vcn_name = "vcn"
+
+tags = {
+  environment = "dev"
+  lob = "finance"
+}
```

Comment:
The example file should have a possible value of how you would configure each parameter. Here the route rules are missing. Since we don't want these to be added, you can add them as commented.

Reply:
`terraform.tfvars.example` contains the root configuration module, but the `custom_route_rule` example don't use variables for module Input `internet_gateway_route_rules` or `nat_gateway_route_rules`. 

Possible values for `internet_gateway_route_rules` or `nat_gateway_route_rules` are directly shown in `main.tf` and explained in detail in `doc/routing_rules.adoc`

In the long run, I don't think initializing every variables in `terraform.tfvars` is the best possible practice. I would keep only provider authentication related variables in this file. Any other variable that needs initialization would be better placed in a `*.auto.tfvars` file. This is also consistent by what is enforced on Terraform Cloud.

But best option in the long run, I would advocate to declare any value to be passed to a module with a Local Value as shown for `internet_gateway_route_rules`. Two advantages here:

1. definition and initialization is contained in a single file (no need to ask the user to create a separate file)
2. we can use functions, Named Values, etc ... which allows for way more possibilities than fixed variable values.

Reply:
will do both to illustrate the options:
- `internet_gateway_route_rules` with variables
- `nat_gateway_route_rules` with locals

And explanation in the README about the preferred way with the advantages.